### ISSUE TITLE
Transparent textures in blender

### ISSUE THREAD

**Issue by BSarmady** [Kruithne/wow.export#392]
Hi, 

I'm not sure if this is reported before. 
I did an export with wow.export-0.1.50 on version 0.1.7.51313 
and picked up a tree (silverpinetree01.m2) to export and import in blender.
In wow.export preview it looks kind of correct (missing leave transparency) But in blender middle of the tree is missing. I checked the normals and they look correct. Tried blend mode = opaque and it looks dark (no texture)

Is this a bug or can be fixed by tweaking the settings in blender or wow.export? 

Wow.export
![image](https://github.com/Kruithne/wow.export/assets/2040435/0fb932dc-d3c2-4ca6-8e38-a31f0ef15655)

Blender
![image](https://github.com/Kruithne/wow.export/assets/2040435/9582d3d6-341c-4912-a8df-7fb03cea809a)

![image](https://github.com/Kruithne/wow.export/assets/2040435/bf640a22-5692-496e-89f8-c01e43255a69)

**Comment by Marlamin** [comment]
The wow.export preview does not support transparency, as for dealing with transparency in Blender [check out the FAQ](https://github.com/Kruithne/wow.export#why-are-my-textures-all-see-through).

**Comment by BSarmady** [comment]
> I think the tree is supposed to be transparent there

Emm, I don't think they are supposed to be floating

I don't have access to wow at the moment, but those trees should look like this.

![image](https://github.com/Kruithne/wow.export/assets/2040435/17b87e93-8a3f-4e93-8d0a-e4bb46dcc2af)
![image](https://github.com/Kruithne/wow.export/assets/2040435/73758082-e5b5-41d2-a722-72221091a6a2)

**Comment by briochie** [comment]
> Emm, I don't think they are supposed to be floating

They are likely back-faces, and you have back-face culling on. Many WoW models, especially old ones, use both back and front faces.

This is still not something wow.export is intended to handle.

**Comment by BSarmady** [comment]
> They are likely back-faces, and you have back-face culling on

Nope, in my 2nd and 3rd screenshots, I am showing normals too. Just to be sure, I turned on and off backface culling too and even checked with face orientation . 

![image](https://github.com/Kruithne/wow.export/assets/2040435/7ec862a8-2b97-4445-837c-a8a52d23b315)

I'll look at the obj code to see if I can figure it out

**Comment by Marlamin** [comment]
Let me know if this is a problem with wow.export and I can re-open it, but there are various things wow.export does not (and will likely not) support so it'll likely end up on that list.

**Comment by BSarmady** [comment]
TL;DR;

This is an assumption, and I'm sure contributors know better than me. I'm assuming wow.export merging object material that are using same texture image to one and then assigning same material to all instances. however this is incorrect as some of those materials require transparency and some do not. Note that this tree is just a sample, I saw this issue with more than one model (while exporting and rendering part of plagueland environment), 

**Long Explanation**

I managed to go to plagueland and found the tree and took screenshot. as you can see the middle part has partially transparent faces.

![WoWScrnShot_091723_161028](https://github.com/Kruithne/wow.export/assets/2040435/772110bf-fb4d-491a-a7dd-e6bee6202046)


After looking at texture, material and model files (PNG, MTL and OBJ) I figured it out.

There are two issues:
1- When converting BLP to PNG we loose image data. Essentially BLP has a separate alpha channel but PNG does not, thus in PNG file, the alpha channel is merged into image, making that part of image transparent even though BLP has image data under the alpha channel area . You can see this by installing BLP plugin for PhotoShop and turning off alpha channel. XnViewMP does not show some BLP and TIFF images correctly.

 <img src="https://github.com/Kruithne/wow.export/assets/2040435/15bb1107-cc39-4a97-b5ef-730badca371c" width="320">
 <img src="https://github.com/Kruithne/wow.export/assets/2040435/8d5071c8-f239-48aa-8532-44e9687290b4" width="320">
 <img src="https://github.com/Kruithne/wow.export/assets/2040435/01977235-ee2f-4584-a8ca-c87e764a077a" width="320">

2- Now in case of tree, in obj file there were 3 textures which is correct, however 2 of them were using same name. **I'm assuming wow export saw them to be using same texture file and replaced both to use same material, however this is incorrect**. First instance of texture does not have transparency (alpha) and second one does. 

To fix this I modified the material file (MTL) with notepad, it is a text file and added a 3rd material changing from

```newmtl mat_silverpinetree01trunkskin
illum 1
map_Kd silverpinetree01trunkskin.png
newmtl mat_silverpinetree01canopyskin
illum 1
map_Kd silverpinetree01canopyskin.png
```
to 
```
newmtl mat_silverpinetree01trunkskin
illum 1
map_Kd silverpinetree01trunkskin.png
newmtl mat_silverpinetree01trunkskin1
illum 1
map_Kd silverpinetree01trunkskin1.png
newmtl mat_silverpinetree01canopyskin
illum 1
map_Kd silverpinetree01canopyskin.png
```
and created two PNG images, one with transparent part and one without.
then modified the obj file (it is also a text file) and renamed the second instance of `mat_silverpinetree01trunkskin` to `mat_silverpinetree01trunkskin1` to reference the one image with transparency.

 I'm sure there is a better way of doing this such as saving image as TIFF and then creating material with separate alpha and data like below to reduce count (and disk size) of textures used

```
newmtl mat_silverpinetree01canopyskin
illum 1
map_Kd silverpinetree01canopyskin.tif
newmtl mat_silverpinetree01trunkskin
illum 1
map_Kd silverpinetree01trunkskin.tif
map_d silverpinetree01trunkskin.tif
newmtl mat_silverpinetree01trunkskin1
illum 1
map_Kd silverpinetree01trunkskin.tif
```

Note the **map_d** in second material

Render outcome of both workarounds was same.

![image](https://github.com/Kruithne/wow.export/assets/2040435/9474a98e-e9e3-46b0-bbe7-60f88f868f69)

So solution would be saving either 
- two PNG files one with alpha merged and one without
- or two PNG one just alpha channel and one image data, 
- or one TIFF file with alpha channel, then create material accordingly.

then creating material file (*.mtl) accordingly.

**Comment by Marlamin** [comment]
Thanks for looking into it, I'll reopen it as a feature request to add one of the suggested options but considering the development status of the program dont get your hopes up in this being fixed.

**Comment by natnat201193** [comment]
It's not exactly important, both trunk geosets use the same texture, one uses alpha, one does not, simple

**Comment by briochie** [comment]
> It's not exactly important, both trunk geosets use the same texture, one uses alpha, one does not, simple

I've noticed this in Unity imports, too. I feel silly for not remembering that now. 🙃 

I would _love_ if we could get materials on a per-material rather than per-texture basis. Would make the metadata much more user-friendly. Right now I'm basically having to guess what material uses what texture.

**Comment by BSarmady** [comment]
> considering the development status of the program dont get your hopes up in this being fixed.

No worries, at least I know how to fix it and in case others came across this issue, they will have one more solution to look into it. The only thing that I think need to be fixed is not replacing material instance in models with a single material to make it is easier to replace alpha channel in blender.
 
> I would _love_ if we could get better information on materials. Right now I'm basically having to guess what material uses what properties in some cases. Color in particular sticks out to me.

Their game engine is simple, as far as I have seen, they do not use color for textures, it is just plain baked texture map and alpha. The only place that might be exception is hair and eye color, which I haven't look at it, but armor and model colors are definitely baked colors into texture itself. I remember a few years back during development of Draenei, they had a 2 hour announcement about being able to paint on landscape and they were very proud of it. 

> It's not exactly important, both trunk geosets use the same texture, one uses alpha, one does not, simple
Correct, I looked at exported png, it seems it has an alpha channel, but it looks like it is not an standard implementation Photoshop and blender both ignore them. This is again speculation, I tried for few hours last night to use alpha channel from png but didn't work, however just replacing the texture with tiff, immediately worked with same settings. It was taking too much time and my intention was just looking at the model to see how they did it. The other thing is export that replacing all instances of material using same texture to single instance which makes it harder to fix in just blender.
